dynamiqs/qarrays/dense_qarray.py

```diff
@@ -195,20 +202,25 @@ def __rmatmul__(self, y: QArrayLike) -> QArray:
             data = y.data @ self.data
         elif isqarraylike(y):
             dims = self.dims
-            data = asjaxarray(y) @ self.data
+            data = _getjaxarray(y) @ self.data
         else:
             return NotImplemented
 
         return DenseQArray(dims, data)
 
-    def __and__(self, y: QArray) -> QArray:
+    def __and__(self, y: QArrayLike) -> QArray:
```

Comment:
This wasn't planned, I don't think we should support `&` with other types than `QArray`. Can we revert? (or do it in another PR, then we also need to fix the base class type)

Reply:
That was a weird one cause we support QArrayLike already in SparseDIAQArray. Should I revert both?

Reply:
Ha I didn't noticed, I propose to revert `SparseDIAQArray` as well. I think we should not overload our newly introduced operators on existing array-like objects.